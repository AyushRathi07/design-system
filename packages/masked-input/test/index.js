import React from 'react'
import TestRenderer from 'react-test-renderer'
import {
  MaskedInput,
  CreditCardInput,
  ExpirationDateInput,
  formatDate,
  formatCreditCard
} from '../src'

const renderJSON = el => TestRenderer.create(el).toJSON()

describe('pcln-masked-input', () => {
  describe('MaskedInput', () => {
    test('renders', () => {
      const json = renderJSON(
        <MaskedInput id="test" value="12/20" format={formatDate} />
      )
      expect(json).toMatchSnapshot()
    })
  })

  describe('formatDate', () => {
    test('formats numbers as MM/YY', () => {
      const date = formatDate('1220')
      expect(date).toBe('12/20')
    })

    test('returns number string with one digit', () => {
      const date = formatDate('1')
      expect(date).toBe('1')
    })

    test('returns empty string with no value', () => {
      const date = formatDate('')
      expect(date).toBe('')
    })
  })

  describe('formatCreditCard', () => {
    test('formats number with hyphens', () => {
      const number = formatCreditCard('4242424242424242')
      expect(number).toBe('4242-4242-4242-4242')
    })

    test('returns empty string with no value', () => {
      const number = formatCreditCard('')
      expect(number).toBe('')
    })
  })

  describe('CreditCardInput', () => {
    test('renders', () => {
      const json = renderJSON(<CreditCardInput id="test" value="4242" />)
      expect(json).toMatchSnapshot()
    })
  })

  describe('ExpirationDateInput', () => {
    test('renders', () => {
      const json = renderJSON(<ExpirationDateInput id="test" value="1020" />)
      expect(json).toMatchSnapshot()
    })
  })
})
