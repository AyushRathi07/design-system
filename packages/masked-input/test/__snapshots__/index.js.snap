// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pcln-masked-input CreditCardInput renders 1`] = `
<input
  autoComplete="cc-number"
  className="sc-jWBwVP civhqK"
  id="test"
  onChange={[Function]}
  placeholder="4242-4242-4242-4242"
  type="text"
  value="4242"
/>
`;

exports[`pcln-masked-input ExpirationDateInput renders 1`] = `
<input
  autoComplete="cc-exp"
  className="sc-jWBwVP civhqK"
  id="test"
  maxLength="5"
  onChange={[Function]}
  placeholder="MM/YY"
  type="text"
  value="1020"
/>
`;

exports[`pcln-masked-input MaskedInput renders 1`] = `
<input
  className="sc-jWBwVP civhqK"
  id="test"
  onChange={[Function]}
  type="text"
  value="12/20"
/>
`;
